crates/control/src/behavior/intercept_ball.rs

```diff
@@ -0,0 +1,45 @@
+use nalgebra::{point, Point2, UnitComplex};
+use types::{
+    HeadMotion, LineSegment, MotionCommand, OrientationMode, PathSegment, PrimaryState, WorldState,
+};
+
+pub fn execute(world_state: &WorldState) -> Option<MotionCommand> {
+    match (
+        world_state.robot.primary_state,
+        world_state.ball,
+        world_state.robot.robot_to_field,
+    ) {
+        (PrimaryState::Playing, Some(ball), Some(robot_to_field)) => {
+            let ball_in_front_of_robot = ball.ball_in_ground.coords.norm() < 2.0
+                && ball.ball_in_ground.x > 0.0
+                && ball.ball_in_ground.y.abs() < 0.5;
+            let ball_moving_towards_robot = ball.ball_in_ground_velocity.x < -0.15;
+            let ball_moving_towards_own_half =
+                (robot_to_field * ball.ball_in_ground_velocity).x < -0.05;
```

Comment:
Done.
Also fixed the thing we talked about where we ignored balls perfect for interception by tried to intercept balls with a far away intercept point just because their current y coordinate was small.

Reply:
Okay, very nice! 😊